app/forms/security_event_form.rb

```diff
@@ -0,0 +1,201 @@
+# rubocop:disable Metrics/ClassLength
+# Handles SET events (Security Event Tokens)
+class SecurityEventForm
+  include Rails.application.routes.url_helpers
+  include ActiveModel::Model
+
+  # From https://tools.ietf.org/html/draft-ietf-secevent-http-push-00#section-2.3
+  module ErrorCodes
+    JWS = 'jws'.freeze
+    JWT_AUD = 'jwtAud'.freeze
+    JWT_CRYPTO = 'jwtCrypto'.freeze
+    JWT_HDR = 'jwtHdr'.freeze
+    JWT_PARSE = 'jwtParse'.freeze
+    SET_DATA = 'setData'.freeze
+    SET_TYPE = 'setType'.freeze
+  end
+
+  validate :validate_iss
+  validate :validate_aud
+  validate :validate_event_type
+  validate :validate_subject_type
+  validate :validate_sub
+  validate :validate_typ
+  validate :validate_exp
+  validate :validate_jwt
+
+  def initialize(body:)
+    @body = body
+    @jwt_payload, @jwt_headers = parse_jwt
+  end
+
+  def submit
+    success = valid?
+
+    if success
+      SecurityEvent.create(
```

Comment:
Is it too cautious to be worried about this call failing, and preferring `create!` instead?

Reply:
Oh that's a great point! I'll update to use that instead